module/zfs/spa_stats.c

```diff
@@ -153,9 +152,7 @@ spa_read_history_init(spa_t *spa)
 	ssh->size = 0;
 	ssh->private = NULL;
 
-	(void) snprintf(name, KSTAT_STRLEN, "zfs/%s", spa_name(spa));
-
-	ksp = kstat_create(name, 0, "reads", "misc",
+	ksp = kstat_create("", 0, "reads", "misc",
```

Comment:
@loli10K the reason for this change is to minimize the stack usage. In kstat create we already have checks for null pointer etc and passing a empty string is just a place holder for the actual name to be assigned later. We can perhaps pass the string (31 chars) that we were passing before which means we keep the original code but reassign the ks_module to a longer name just before calling kstat_install (as per the new code).